integration/network/dns_test.go

```diff
@@ -102,8 +102,10 @@ func TestExtDNSInIPv6OnlyNw(t *testing.T) {
 	network.StartDaftDNS(t, "127.0.0.1")
 
 	// Set up a temp resolv.conf pointing at that DNS server, and a daemon using it.
-	tmpFileName := network.WriteTempResolvConf(t, "127.0.0.1")
-	d := daemon.New(t, daemon.WithEnvVars("DOCKER_TEST_RESOLV_CONF_PATH="+tmpFileName), daemon.WithExperimental())
+	d := daemon.New(t,
+		daemon.WithResolvConf(network.WriteTempResolvConf(t, "127.0.0.1")),
+		daemon.WithExperimental(),
+	)
```

Comment:
Heh,  this now makes it really tempting to combine those (not sure if we should), as it looks like `WriteTempResolvConf` is always used in combination with `daemon.WithResolvConf`

```go
daemon.WithResolver(t, "127.0.0.1")
```

There is some special handling in that utility for permissions; https://github.com/moby/moby/blob/f95e4b3d22260eb3322b4be13d391b8db2d4199f/integration/internal/network/dns.go#L20-L24


I wonder if (as we're passing the `Daemon` struct to these options) if putting it somewhere inside the daemon's storage location would work for that; in that case, we'd keep all things related to that daemon instance together;
https://github.com/moby/moby/blob/f95e4b3d22260eb3322b4be13d391b8db2d4199f/testutil/daemon/daemon.go#L200-L204

https://github.com/moby/moby/blob/f95e4b3d22260eb3322b4be13d391b8db2d4199f/testutil/daemon/daemon.go#L106-L122

Reply:
> Heh, this now makes it really tempting to combine those (not sure if we should), as it looks like WriteTempResolvConf is always used in combination with daemon.WithResolvConf

That'd take away the possibility of giving the daemon a more complicated `resolv.conf` (or a deliberately broken one).

I thought about having `StartDaftDNS` write its own `resolv.conf`, returning the filename. Then, in four of the five cases, it'd be ... `daemon.WithResolvConf(network.StartDaftDNS(t, "127.0.0.1"))`. (In the other case, no dummy resolver gets started, the test just needs its own `resolv.conf`.)

But, same thing there really - it'd limit the `resolv.conf`, without buying much.

> I wonder if (as we're passing the Daemon struct to these options) if putting it somewhere inside the daemon's storage location would work for that; in that case, we'd keep all things related to that daemon instance together;

I suppose `WithResolvConf` could take a `[]byte` to write to a `resolv.conf` in the daemon's `Folder`, instead of the location of a file (so the file doesn't need to be created before the `NewDaemon` has created `Folder`). I'll try that...

Reply:
> I suppose WithResolvConf could take a []byte to write to a resolv.conf in the daemon's Folder, instead of the location of a file (so the file doesn't need to be created before the NewDaemon has created Folder). I'll try that...


Oh, that might work yes!


And, yes, I was also considering "would we need more complex ones?" so yeah, I can agree.

It's mostly brainstorming a bit here, knowing that this is not a critical/urgent PR, so just looking what shape it could take to make it clean and flexible enough 😄 ❤️